src/apps/activity_assignments/crud/assignments.py

```diff
@@ -78,38 +79,39 @@ async def create_many(self, schemas: list[ActivityAssigmentSchema]) -> list[Acti
 
         return await self._create_many(schemas)
 
-    async def already_exists(self, schema: ActivityAssigmentSchema) -> ActivityAssigmentSchema:
+    async def exist(self, assignment: ActivityAssignmentCreate) -> ActivityAssigmentSchema | bool:
         """
-        Checks if an activity assignment already exists in the database.
+        Checks if an activity assignment exists in the database.
 
         This method builds a query to check for the existence of an assignment with the same
         `activity_id`, `activity_flow_id`, `respondent_subject_id`, and `target_subject_id`,
         while ensuring the record has not been soft-deleted.
 
         Parameters:
         -----------
-        schema : ActivityAssigmentSchema
+        assignment : ActivityAssignmentCreate
             The activity assignment schema to check for existence.
 
         Returns:
         --------
-        bool
-            `True` if the assignment already exists, otherwise `False`.
+        ActivityAssigmentSchema | bool
+            The value of the first matching record, if it exists. Otherwise, returns False.
 
         Notes:
         ------
-        - This method uses the `_execute` method from the `BaseCRUD` class to run the query and
-        check for the existence of the assignment.
-        - The existence check is based on the combination of IDs and considers soft-deleted records.
+        - The existence check excludes soft-deleted records.
         """
         query: Query = select(ActivityAssigmentSchema)
-        query = query.where(ActivityAssigmentSchema.activity_id == schema.activity_id)
-        query = query.where(ActivityAssigmentSchema.respondent_subject_id == schema.respondent_subject_id)
-        query = query.where(ActivityAssigmentSchema.target_subject_id == schema.target_subject_id)
-        query = query.where(ActivityAssigmentSchema.activity_flow_id == schema.activity_flow_id)
+        query = query.where(
+            ActivityAssigmentSchema.activity_id == assignment.activity_id,
+            ActivityAssigmentSchema.activity_flow_id == assignment.activity_flow_id,
+            ActivityAssigmentSchema.respondent_subject_id == assignment.respondent_subject_id,
+            ActivityAssigmentSchema.target_subject_id == assignment.target_subject_id,
+            ActivityAssigmentSchema.soft_exists(),
+        )
 
         db_result = await self._execute(query)
-        return db_result.scalars().first()
+        return db_result.scalars().first() or False
 
```

Comment:
we have first() and one_or_none(), both are the same, but second seems more clear of the return if your function returns an object or None